### PR TITLE
app: add memory monitor

### PR DESCRIPTION
**Pull request by corverroos**
Adds a memory monitor that writes heap dumps to disk when memory increases. This aims to catch suspicious spikes seen in the wild.

category: misc
ticket: none

### CODE REVIEW

**Comment by corverroos** [comment]
image

File: app/heap.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			stats := new(runtime.MemStats)
+			runtime.ReadMemStats(stats)
+			if stats.HeapInuse < threshold {
+				continue
+			}
+
+			filename := fmt.Sprintf("heap_dump_%d.pb.gz", time.Now().Unix())
```
Comment (dB2510) [inline app/heap.go:74]: this will create alot of files

File: app/lifecycle/order.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ const (
 	StartScheduler
 	StartP2PEventCollector
 	StartPeerInfo
+	StartRest
```
Comment (dB2510) [inline app/lifecycle/order.go:41]: why not `StartMemMonitor`?

**Review by dB2510 — APPROVED** [review]

**Comment by corverroos** [comment]
The memory issues were due to jaeger storing tracing in memory.

**Pull request closed without merging**